amazon-s3/6.0/modules/ROOT/pages/amazon-s3-connector-reference.adoc

```diff
@@ -0,0 +1,3652 @@
+
+To access an archived object, you must restore the object for the duration (number of days) that you specify.
+
+To restore a specific object version, you can provide a version ID. If you don't provide a version ID, Amazon S3 restores the current version.
+
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Bucket name a| String |  Bucket name containing the object to restore. |  | x
+| Object key a| String |  Object key for which the action is initiated. |  | x
+| Version Id a| String |  Version ID used to reference a specific version of the object. |  |
+| Days a| Number |  Lifetime of the active copy in days. Do not use with restores that specify OutputLocation. |  | x
+| Glacier Job Tier a| Enumeration, one of:
+
+** Standard
+** Bulk
+** Expedited |  Retrieval tier at which the restore is processed. |  | x
+| Config Ref a| ConfigurationProvider |  Name of the configuration to use to execute this component. |  | x
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  `#[payload]` |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+==== Output
+
+[%autowidth.spread]
+|===
+|Type |<<RestoreObjectResponse>>
+|===
+
+==== Associated Configurations
+
+* <<config>>
+
+==== Throws
+
+* S3:BAD_REQUEST
+* S3:CONNECTIVITY
+* S3:FORBIDDEN
+* S3:GLACIER_EXPEDITED_RETRIEVAL_NOT_AVAILABLE
+* S3:INVALID_OBJECT_STATE
+* S3:NO_SUCH_KEY
+* S3:OBJECT_ALREADY_IN_ACTIVE_TIER
+* S3:RESTORE_ALREADY_IN_PROGRESS
+* S3:RETRY_EXHAUSTED
+* S3:S3_ERROR
+* S3:SERVER_BUSY
+* S3:SERVICE_ERROR
+
+
+[[SelectObjectContent]]
+=== Select Object Content
+`<s3:select-object-content>`
+
+
+Filters the contents of an Amazon S3 object based on a simple structured query language (SQL) statement.
+
+In the request, along with the SQL expression, you must also specify a data serialization format (JSON, CSV, or Apache Parquet) of the object. Amazon S3 uses this format to parse object data into records, and returns only records that match the specified SQL expression.
+
+You must also specify the data serialization format for the response. You can use Amazon S3 Select to query objects that have the following format properties: CSV, JSON, or Parquet. UTF-8 is the only encoding type Amazon S3 Select supports. CSV and JSON files can be compressed using GZIP or BZIP2. GZIP and BZIP2 are the only compression formats that Amazon S3 Select supports for CSV and JSON files. Amazon S3 Select supports columnar compression for Parquet using GZIP or Snappy. Amazon S3 Select does not support whole-object compression for Parquet objects.
```

Comment:
You must also specify the data serialization format for the response. You can use Amazon S3 Select to query objects formatted as CSV, JSON, or Parquet. 

Amazon S3 Select supports the following compression types:
GZIP and BZIP2 for CSV and JSON files
Columnar compression for Parquet using GZIP or Snappy.
+
Amazon S3 Select does not support whole-object compression for Parquet objects.